### PR TITLE
emergency release in place, tests pass

### LINKED ISSUE #28: emergency refactor

### PR DESCRIPTION
**Pull request by shaharyakir**
Closes #28

### CODE REVIEW

File: contracts/insured-vesting-v1/InsuredVestingV1.sol [hunk 1]
```diff
@@ -63,12 +64,19 @@ contract InsuredVestingV1 is Ownable {
...
+    error AlreadyEmergencyReleased();
+    error OnlyOwnerOrSender();
 
     modifier onlyBeforeVesting() {
         if (startTime != 0 && block.timestamp > startTime) revert VestingAlreadyStarted();
         _;
     }
 
+    modifier onlyOwnerOrSender(address target) {
+        if (msg.sender != owner() && msg.sender != target) revert OnlyOwnerOrSender();
```
Comment (Luke-Rogerson) [inline contracts/insured-vesting-v1/InsuredVestingV1.sol:76]: I think this is wrong, no? You want to enforce the user is **either** the owner **or** an investor, but you are currently enforcing they are both (`&&`)?

Comment (shaharyakir) [reply contracts/insured-vesting-v1/InsuredVestingV1.sol:76]: the NOT is what makes the change - (if you're not ALLOWED_USER_1 AND you're not ALLOWED_USER_2) then you're not allowed

File: contracts/insured-vesting-v1/InsuredVestingV1.sol [hunk 2]
```diff
@@ -170,49 +177,43 @@ contract InsuredVestingV1 is Ownable {
...
         }
 
-        IERC20(tokenAddress).safeTransfer(owner(), tokenBalanceToRecover);
+        IERC20(tokenAddress).safeTransfer(project, tokenBalanceToRecover);
 
         // in case of ETH, transfer the balance as well
-        Address.sendValue(payable(owner()), address(this).balance);
+        Address.sendValue(payable(project), address(this).balance);
 
+        // TODO why twice..?
```
Comment (Luke-Rogerson) [inline contracts/insured-vesting-v1/InsuredVestingV1.sol:214]: What do you mean here?

Comment (shaharyakir) [reply contracts/insured-vesting-v1/InsuredVestingV1.sol:214]: we transfer eth twice for some reason

File: contracts/insured-vesting-v1/InsuredVestingV1.sol (hunk 1 above)
Comment (shaharyakir) [reply contracts/insured-vesting-v1/InsuredVestingV1.sol:76]: this could be rewritten as `if !(msg.sender == owner() || msg.sender == target)`

**Review by Luke-Rogerson — APPROVED** [review]

**Pull request merged**